File: code/vision/branches/20090415/src/java/cast/server/ComponentServer.java
/**
 * 
 */
package cast.server;

import Ice.Application;
import Ice.Communicator;
import Ice.Identity;
import Ice.ObjectAdapter;
import Ice.SignalPolicy;
import cast.cdl.JAVASERVERPORT;

/**
 * @author nah
 * 
 */
public class ComponentServer extends Application {

	/**
	 * 
	 */
	public ComponentServer() {
	}

	/**
	 * @param _signalPolicy
	 */
	public ComponentServer(SignalPolicy _signalPolicy) {
		super(_signalPolicy);
	}

	/*
	 * (non-Javadoc)
	 * 
	 * @see Ice.Application#run(java.lang.String[])
	 */
	@Override
	public int run(String[] arg0) {
		Communicator ic = communicator();
		ObjectAdapter adapter = ic.createObjectAdapterWithEndpoints(
				"ComponentServer1", "default -p " + JAVASERVERPORT.value);

		Identity id = new Identity("ComponentFactory", "ComponentFactory");
		adapter.add(new CASTComponentFactory(), id);
		
		
		Identity manid = new Identity("comp.man", CASTComponentManager.class
				.getCanonicalName());
		adapter.add(new CASTComponentManager(), manid);

		
		//		adapter.addServantLocator(new cast.server.ComponentLocator(), "");
		adapter.activate();
		adapter.waitForDeactivate();
		return 0;
	}

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		ComponentServer app = new ComponentServer();
		int status = app.main("cast.server.ComponentServer", args);
		System.exit(status);
	}
}

File: code/comsys/tags/release1/tools/cast/src/c++/cast/server/ComponentServer.cpp

#include <CASTComponentFactory.hpp> 
#include <CASTTimeServer.hpp> 

#include <CDL.hpp> 
#include <Ice/Ice.h> 
 
using namespace Ice;
using namespace std;

namespace cast {
  
  class ComponentServer : virtual public Ice::Application { 
  public: 
    virtual int run(int, char*[]) { 
     
      CommunicatorPtr ic = communicator();
      
      char buf[50];
      snprintf(buf,50,"default -p %d", cast::cdl::CPPSERVERPORT);
      
      ObjectAdapterPtr adapter 
	= ic->createObjectAdapterWithEndpoints("ComponentServer1", buf);
      
      //add component factory to server
      Identity factoryID;
      factoryID.name = "ComponentFactory";
      factoryID.category = "ComponentFactory";
      adapter->add(new cast::CASTComponentFactory(), factoryID);
      
      //add timeserver to server
      
      Identity tsID;
      tsID.name = "TimeServer";
      tsID.category = "TimeServer";
      adapter->add(new cast::CASTTimeServer(), tsID);
      
      
      adapter->activate();
      
      ic->waitForShutdown();
      
      return 0; 
    }

  };

}; 
  
int 
main(int argc, char* argv[]) {
  cast::ComponentServer app; 
  try {
    return app.main(argc, argv ); 
  }
  catch(...) {
    cout<<"Exception in ComponentServer"<<endl;
    return 1;
  }
}
